atuin-client/src/settings.rs

```diff
@@ -303,6 +304,107 @@ impl Default for Stats {
     }
 }
 
+#[derive(Clone, Debug, Default, Deserialize)]
+pub struct Styles {
```

Comment:
I know this isn't the only thing here, but would you be open to the following

1. Split settings into a module
2. Break the style stuff into it's own file

Don't worry about the rest for now, but `settings.rs` is getting fairly unwieldy now imo!

Reply:
> When you say another module, are you thinking about a settings.rs as a root, containing only top-level construction-type code and then a settings/ dir containing things like styles.rs for example?

Exactly that!

And yes, the re-org I was considering is more-or-less what you say. However, I'd also like to break all "history" specific stuff out into an `atuin-history` crate. Then, we can make the main `atuin` binary just connect a few things together. Kinda like `atuin-config`.

Still a bit of a fuzzy idea though, and I'd like to see how other things shake out first. I'd also like to consider moving to KDL for config.

(gone way beyond the scope of this PR now, but just context setting)

Reply:
That makes sense, I'll have a quick play around and see how dramatic it would be to include that in this set of changes. Just taking the struct members out shouldn't mean any changes to consumers, if we wanted to group some of the current settings into new types (e.g. paths) then code consumers would need updating and that might get quite extensive (shouldn't have a problem with the config file itself if new structs are serde flattened). Will see how it looks!

Reply:
Ok, I've started playing with a bit of a settings refactor, which can be seen in the latest commit. What I've done for now...
* Pulled groups of settings out into submodules of settings with vaguely sensible names (I hope) and any specialized types where relevant
* Re-exported those types from the original settings module, to avoid needing to change other code

What I've also done for one of those sets of properties (the `display` submodule) is to pull the properties into a new settings type at that submodule level, as well as a function for setting defaults for that submodule on the `ConfigBuilder` instance. I've then replaced the raw properties on settings with the new settings type, but with a `serde` attribute to flatten the type, so the structure of the actual config file is unchanged. That obviously does need some changes elsewhere, so those changes are reflected wherever there is now a "`.display`" included in a settings path.

The new submodule settings type is a potential option, just to see how well that fits in. I don't have any strong opinions on that one, see what you think and whether it's worth the slight extra code.